misc/include/naive_rmm_tree.h

```diff
@@ -0,0 +1,274 @@
+#pragma once
+#include <vector>
+#include <string>
+#include <cstdint>
+#include <limits>
+#include <algorithm>
+
+struct NaiveRmM
```

Comment:
In modern C++ struct is supposed to be used if it is a simple structure where you just want to access it's fields and want to name them rather than use something like `pair` or `tuple`, may be with few methods. Here you have a lot of methods and it is convenient to make it a class a decide what is accessible from the outside of the class and what is not. From what i see it is fine to make all the methods private and access only via methods.